### DISCUSSION TITLE
Difference between `|` and `+`?

### DISCUSSION THREAD

**Discussion by koudelka** [authzed/spicedb#1513] [Q&A] [answered]
Is there an actual difference between the `|` and`+` operators? 

The `+` operator has [documentation](https://authzed.com/docs/reference/schema-lang#-union), but the `|` doesn't.

The schema playground seems to allow them to be used interchangeably.

**Comment by josephschorr** [comment, marked as answer]
`|` is for specifying different possible subject types on a `relation` and cannot be used as part of a `permission`: `relation foo: firsttype | secondtype`

`+` is used for combining different relations within a `permission` and cannot be used on a `relation`: `permission view = viewer + editor`.

They are not interchangeable

**Comment by koudelka** [comment]
Ah, you're totally right. I didn't see the little red syntax-error squiggle, color perception issues!

**Reply by josephschorr** [reply]
If you have any suggestions on making that more visible, please let us know
